fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/FedoraEventImpl.java

```diff
@@ -227,46 +224,68 @@ public String toString() {
      */
     public static EventType valueOf(final Integer i) {
         final EventType type = translation.get(i);
-        if (isNull(type)) {
+        if (type == null) {
             throw new IllegalArgumentException("Invalid event type: " + i);
         }
         return type;
     }
 
-
     /**
      * Convert a JCR Event to a FedoraEvent
+     *
      * @param event the JCR Event
      * @return a FedoraEvent
      */
     public static FedoraEvent from(final Event event) {
         requireNonNull(event);
         try {
-            @SuppressWarnings("unchecked")
             final Map<String, String> info = event.getInfo();
-
-            return new FedoraEventImpl(valueOf(event.getType()), cleanPath(event),
+            final FedoraEventImpl fedoraEvent = new FedoraEventImpl(valueOf(event.getType()), cleanPath(event),
                     event.getUserID(), event.getUserData(), event.getDate(), info);
-
+            getResourceTypes(event).forEach(fedoraEvent::addResourceType);
```

Comment:
The namespaces for these resource types need to be dereferenced in a manner similar to how they were handled with the (removed) namespaceResolver function. The problem is that types take this form internally `fedora:Resource`, but when published on the eventBus, they need to be `http://fedora.info/definitions/v4/repository#Resource`